php/blocks/controls/class-repeater.php

```diff
@@ -63,4 +63,31 @@ public function register_settings() {
 			)
 		);
 	}
+
+	/**
+	 * Remove empty placeholder rows.
+	 *
+	 * @param mixed $value The value to either make available as a variable or echoed on the front-end template.
+	 * @param bool  $echo Whether this will be echoed.
+	 * @return mixed $value The value to be made available or echoed on the front-end template.
+	 */
+	public function validate( $value, $echo ) {
+		if ( isset( $value['rows'] ) ) {
```

Comment:
How about something like:

```php
array_filter( $value['rows'] );
```

It [looks like](https://www.thecodedeveloper.com/remove-empty-array-elements-in-php/) that removes empty array values, like `'' => ''`

Reply:
This would be nice, except that `array_filter` only filters empty values.

We could supply a custom callback to array_filter which targets empty keys, or alternatively use something like this:
```
array_diff_key( $value['rows'][ $key ], array_flip( array( '', 0 ) ) );
```

but ultimately, I think just unsetting the two is a better, simpler, solution.

Reply:
Ah, it removes empty values. That's not good.